### PR TITLE
[UR][doc] Add contibution guide

### PR DESCRIPTION
**Pull request by veselypeta**
Contribution Guide documents:

* The processes we use to contribute changes to the specification
* The requirements for having a change accepted
* Some requirements for testing

Additions welcome.

### CODE REVIEW

File: scripts/core/CONTRIB.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+Forks and Pull Requests
+=======================
+
+To submit a pull request to Unified Runtime, you must first create your own 
+personal fork of the project and submit your changes to a branch. By convention 
+we name our branches ``<your_name>/<short_description>``, where the description 
+indicates the intent of your change. You can then raise a pull request targeting 
+``oneapi-src/unified-runtime:main``.
+
+Before your pull request is merged it `must` be reviewed by no less than two
```
Comment (martygrant) [inline scripts/core/CONTRIB.rst:56]: More one for the repo admins but I'm sure it lets me merge with less than two approvals? Or it at least appears that way.

File: scripts/core/CONTRIB.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+personal fork of the project and submit your changes to a branch. By convention 
+we name our branches ``<your_name>/<short_description>``, where the description 
+indicates the intent of your change. You can then raise a pull request targeting 
+``oneapi-src/unified-runtime:main``.
+
+Before your pull request is merged it `must` be reviewed by no less than two 
+code owners of the project `and` be passing all required CI jobs.
+
+
+Conformance Testing
```
Comment (martygrant) [inline scripts/core/CONTRIB.rst:60]: Might be worth expanding a little to say how the tests get run?

**Review by martygrant — APPROVED** [review]
LGTM

File: scripts/core/CONTRIB.rst [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+Naming Conventions
+~~~~~~~~~~~~~~~~~~
+
+Additional to what is described in the Yaml syntax document. You should follow
+the following naming convention for argument names.
+
+    * Entry-Points use ``camelCase`` conventions i.e ``${x}DeviceGet``.
+
+    * Argument names names are also `camelCase` are prefixed with a type
```
Comment (pbalcer) [inline scripts/core/CONTRIB.rst:36]:
```suggestion
    * Argument names are also `camelCase` and are prefixed with a type 
```

File: scripts/core/CONTRIB.rst (hunk 1 above)
Comment (pbalcer) [reply scripts/core/CONTRIB.rst:56]: Yes, we don't have the repo configured to strictly enforce this rules (and I'm not sure if we should).

**Review by pbalcer — APPROVED** [review]

File: scripts/core/CONTRIB.rst (hunk 1 above)
Comment (veselypeta) [reply scripts/core/CONTRIB.rst:56]: I think this is fine for us to make exceptions. This is mostly for external or outside contributors.

File: scripts/core/CONTRIB.rst (hunk 2 above)
Comment (kbenzie) [reply scripts/core/CONTRIB.rst:60]: I think this is worthwhile info but I don't think this needs to block this PR. Could you create an issue for this @martygrant and link it back here?

File: scripts/core/CONTRIB.rst [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+Conformance Testing
+===================
+
+For contributions to the specification you should include appropriate 
+conformance tests as part of your change. The conformance tests can be found 
+under ``test/conformance/<component>``, where component refers to a set of 
+entry-points i.e. platform, enqueue, device. The conformance tests, at a 
+minimum, should cover at least one test for each of the possible error codes 
+returned, excluding any disaster cases like 
+``${X}_RESULT_ERROR_OUT_OF_HOST_MEMORY`` or similar.Conformance tests should not
```
Comment (kbenzie) [inline scripts/core/CONTRIB.rst:69]:
```suggestion
``${X}_RESULT_ERROR_OUT_OF_HOST_MEMORY`` or similar. Conformance tests should not
```

**Review by kbenzie — APPROVED** [review]

File: scripts/core/CONTRIB.rst (hunk 2 above)
Comment (veselypeta) [reply scripts/core/CONTRIB.rst:60]: https://github.com/oneapi-src/unified-runtime/issues/516

**Pull request merged**